src/settings/toml/deploy_target.rs

```diff
@@ -0,0 +1,155 @@
+use crate::settings::toml::Route;
+
+#[derive(Debug)]
+pub struct RouteConfig {
+    pub workers_dev: Option<bool>,
+    pub route: Option<String>,
+    pub routes: Option<Vec<String>>,
+    pub zone_id: Option<String>,
+    pub account_id: Option<String>,
+}
+
+impl RouteConfig {
+    fn has_conflicting_targets(&self) -> bool {
+        if self.workers_dev.unwrap_or_default() {
+            self.routes_defined()
+        } else {
+            if let Some(pattern) = &self.route {
```

Comment:
i think it's a bit confusing to switch between `route` and `pattern` as nouns here - they should be consistent.

Reply:
we will have to use `pattern` when we make the api call, but i can try to shift that logic